p2p/dataset.go

```diff
@@ -14,7 +14,7 @@ import (
 const MtDatasetInfo = MsgType("dataset_info")
 
 // RequestDataset fetches info about a dataset from qri peers
-// It's expected the local peer has attempted to canonicalize the reference
+// It's expected the local peer has attempted to resolve the reference
 // before sending to the network
 // ref is used as an outparam, populating with data on success
 func (n *QriNode) RequestDataset(ctx context.Context, ref *reporef.DatasetRef) (err error) {
```

Comment:
Oh boy is this method out of date!

First, we have the comment  `It's expected the local peer has attempted to resolve the reference`, and then we're calling `ResolveRefLocally` within the method... yeah.

But before we go fixing that, this method has zero callers within the codebase (other than a test). mainly because this logic has been split & superseded by two new concepts:

* reference resolution
* dataset previews

Let's remove this method entirely 😄